### PR TITLE
Hotfix/unique index

### LINKED ISSUE #117: Bug: Folder uploading - unique index
Can't upload folders more than once.
Folders have an empty `key` value by definition.
error: `InvalidArgument desc = unique index 'key : { "" }' is already in use`

### PR DESCRIPTION
**Pull request by Shahar-Y**
fixed indexing, connection string in tests. folders don't have key and bucket. added tests and fixed tests.
closes #117

### CODE REVIEW

File: src/file/file.service.ts [hunk 1]
```diff
@@ -46,30 +46,29 @@ export class FileService {
...
       size,
-      _id: id,
-      parent: parentID,
-    });
+      parent: parentID
+    };
+
+    // Create the file id by reversing key, and add ket and bucket.
+    if (key) {
+      id = this.reverseString(key);
```
Comment (avivbarel) [inline src/file/file.service.ts]: Create a function `generateKey()/generateS3Key()` or whatever you find suitable and use it here.

Comment (avivbarel) [reply src/file/file.service.ts]: There's no need for the comment if you have a function for it, and even more if the function has it's own comment.

File: src/file/file.service.ts [hunk 2]
```diff
@@ -46,30 +46,29 @@ export class FileService {
 
     let id: string | ObjectID = new ObjectID();
 
-    // Create the file id by reversing key.
-    if (key) {
-      id = this.reverseString(key);
-    }
-
     // If there is no parent given, create the file in the user's root folder.
     const parentID: string = folderID;
```
Comment (avivbarel) [inline src/file/file.service.ts]: Is this needed?

File: src/file/file.service.ts (hunk 1 above)
Comment (Shahar-Y) [reply src/file/file.service.ts]: done

File: src/file/file.service.ts (hunk 2 above)
Comment (Shahar-Y) [reply src/file/file.service.ts]: done

**Pull request merged**